llama_index/response_synthesizers/refine.py

```diff
@@ -224,12 +228,16 @@ def _refine_response_single(
         for cur_text_chunk in text_chunks:
             query_satisfied = False
             if not self._streaming:
-                structured_response = cast(
-                    StructuredRefineResponse, program(context_msg=cur_text_chunk)
-                )
-                query_satisfied = structured_response.query_satisfied
-                if query_satisfied:
-                    response = structured_response.answer
+                try:
+                    structured_response = cast(
+                        StructuredRefineResponse, program(context_msg=cur_text_chunk)
+                    )
+                    query_satisfied = structured_response.query_satisfied
+                    if query_satisfied:
+                        response = structured_response.answer
+                except Exception as e:
+                    logger.error(f'Error: {e}', exc_info=True)
+                    return response
             else:
```

Comment:
Skipping if issue occurs to process the next chunk, there are async functions as well have updated those too